monitor/census.go

```diff
@@ -2028,8 +2030,28 @@ func AIGPUsIdle(currentGPUsIdle int) {
 	stats.Record(census.ctx, census.mAIGPUsIdle.M(int64(currentGPUsIdle)))
 }
 
-func AICurrentLiveSessions(currentPipelines int) {
-	stats.Record(census.ctx, census.mAICurrentLivePipelines.M(int64(currentPipelines)))
+func AICurrentLiveSessions(sessionsByPipeline map[string]int) {
+	census.lock.Lock()
+	defer census.lock.Unlock()
+
+	// Reset all existing pipeline live session counts to zero first.
+	// This ensures that pipelines that no longer have live sessions are correctly reported as 0.
+	for k := range census.aiLiveSessionsByPipeline {
+		census.aiLiveSessionsByPipeline[k] = 0
```

Comment:
Maybe we could just send a zero metric and then remove this pipeline from the map completely if it's not reporting any new sessions. 

We should avoid constantly sending `0` for dead pipelines and avoid keeping it in memory endlessly